students/morgan/Mailroom/mailroom4.py

```diff
@@ -0,0 +1,215 @@
+import math
+import functools
+
+
+
+infile = 'donors.txt'
+donor_objects = []
+
+
+def read_donors():
+    master_donor = []
+    
+    with open(infile) as donor_input:
+        donor_input = [w for line in open(infile) for w in line.rstrip('\n').split("\n")]
+        donor_input.remove("Donor Name - Donations with comma dilimiter")
+
+        for line in donor_input:
+            name_string = line.split('-')[0].strip()
+            donation_string = line.split('-')[1].strip(" ").strip()
+            donation_list = donation_string.split(',')
+
+            donation_list = [float(x) for x in donation_list]
+
+
+            temp = [name_string, donation_list]
+            master_donor.append(temp)
+            #donors[name_string.lower()] = [name_string, donation_list]
+    
+    
+    
+    # print(master_donor) 
+
+    for i, v in enumerate(master_donor):
+        # print(i)
+        # print(str(temp_name))
+        name = str(master_donor[i][0])
+        # print(name)
+        donation = master_donor[i][1]
+        # print(donation)
+        temp_name = Donor(name, donation)
+        # print(temp_name.name, temp_name.donation)
+        
+        donor_objects.append(temp_name)
+        DB.add_donor(temp_name)
+
+
+
+
+        # donation = list(master_donor[i][1])
+        # print(temp_name)
+        # temp_name = Donor(name, donation)
+        # print(temp_name.name)
+        # print(temp_name.donation)
+        # master_donor.append(temp_name)
+
+
+
+
+
+
+class Donor(object):
+    
+    # name = ""
+    # donations = []
+    def __init__ (self, name, donations):
+        
+        self.name = name
+        self.donations = donations
+
+    def add_donation(self, money):
+
+        self.donations.append(money)
+
+    @property
+    def total(self):
+        return sum(self.donations)
+
+    @property    
+    def maxdonation(self):
+        return max(self.donations)
+
+        
+
+    @property
+    def namelength(self):
```

Comment:
not  muych point to this -- a_donor.name is expected to be a string -- so if you want to know how long it is, you can simply so:

len(a_donor.name)

you want a property to "hide" implementation details -- but the name attribute being a string is part of the API, not an implementation detail.

Reply:
The other properties are great -- they hide (or encapsulate) the fact that the donations are stored as a simple list.